examples/pytorch/AxHyperOptimizationPTL/AxHyperOptimizationPTL.py

```diff
@@ -0,0 +1,85 @@
+import argparse
+import mlflow
+from ax.service.ax_client import AxClient
+from iris import IrisClassification
+from iris_data_module import IrisDataModule
+import pytorch_lightning as pl
+
+
+def train_evaluate(params, max_epochs=100):
+    model = IrisClassification(**params)
+    dm = IrisDataModule()
+    dm.prepare_data()
+    dm.setup(stage="fit")
+    trainer = pl.Trainer(max_epochs=max_epochs)
+    mlflow.pytorch.autolog()
+    trainer.fit(model, dm)
+    trainer.test(datamodule=dm)
+    test_accuracy = trainer.callback_metrics.get("test_acc")
+    return test_accuracy
+
+
+def model_training_hyperparameter_tuning(max_epochs, total_trials, params):
+    """
+     This function takes input params max_epcohs.epxeirment_name,total_trials,params and tracking_uri
+     and creates a nested run in Mlflow.  The parameters,metrics,model and summary are dumped into their
+     respective mlflow-run ids. The best parameters are dumped along with the basedline model.
+
+    :param max_epochs:Max epochs used for training the model. Type:int
+    :param experiment_name: Mlflow experiment name , in which the runs would be logged. Type:str
+    :param total_trials: Number of ax-client experimental trials. Type:int
+    :param params: Model parameters. Type:dict
+    :param tracking_uri: Mlflow tracking_uri
+    """
+    mlflow.start_run(run_name="Parent Run")
+    train_evaluate(params=params, max_epochs=max_epochs)
```

Comment:
Can we remove `train_evaluate` here?

Reply:
`train_evaluate` must remain as it trains the base model.

Reply:
What is the base model?

Reply:
The model before starting the pruning process is the base model.

Reply:
I think logging the base model with the best params in the same run seems a bit confusing. Can we log the base model in a separate run?